### ISSUE TITLE
Yoda comparison not flagged / new testcase

### ISSUE THREAD

**Issue by jrfnl** [WordPress-Coding-Standards/WordPress-Coding-Standards#497]
The below code did not get flagged, not sure why (haven't looked in depth).

```php
if ( $GLOBALS['wpdb']->num_rows === 0 ) {}
```

**Comment by JDGrimes** [comment]
This line: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/WordPress/Sniffs/PHP/YodaConditionsSniff.php#L78

```php
			// If this is a function call or something, we are OK.
			if ( in_array( $tokens[ $i ]['code'], array( T_STRING, T_CLOSE_PARENTHESIS, T_OPEN_PARENTHESIS, T_RETURN ) ) ) {
				return;
			}
```

It bails on `T_STRING`, which is how the `num_rows` above would be classified. I *think* that we could just remove `T_STRING` from that list, but I'm not 100% sure.

**Comment by jrfnl** [comment]
Found another one which didn't get flagged:
```php
if( trim( $this->dirty->name ) !== '' ) {}
```

**Comment by JDGrimes** [comment]
> Edit: then again, I suppose this one shouldn't necessarily be flagged as it's a function call. Still, would make sense if it did get flagged for code consistency.

Yeah, previously the decision was made to only check cases where there is a variable and so an assignment could accidentally take place. I think it makes sense to be consistent across the board, but the problem is determining which side of the condition should come first. Like in your example above, we'd have to analyze both sides of the condition and then make an educated guess. It would make the sniff much more complex. (This particular case could be checked for easily, but then there will be another and another and...)
